### DISCUSSION TITLE
Suggested release plan...

### DISCUSSION THREAD

**Discussion by lovelydinosaur** [encode/django-rest-framework#9516] [General]
Hey team. We still have a couple of remaining issues that we ought to address to get back to a neater clearer and more stable release process. Suggested release roadmap from here...

---

### REST framework 3.15.3

*(Date: ASAP)*

* Revert or otherwise address UniqueConstraint validator regressions. (#9358, #9378, #9410, #9484)
* Keep same Python/Django versions as 3.15.1. (#9450)

### REST framework 3.16.0

*(Date: ASAP)*

* Support Django 5.1. (#9514)
* Drop support for Django < 4.2 and Python < 3.8. (#9393)
* Drop deprecated `AutoSchema._get_reference` method. Currently marked as `RemovedInDRF316Warning`.

### REST framework 3.17.0

*(Date: T+6 months)*

* Support Django 5.2
* Drop `coreapi` support throughout codebase. Currently marked as `RemovedInDRF317Warning`.

---

*(Onwards... continue with a 6 monthly release cycle in line with new Django releases)*

**Comment by browniebroke** [comment]
As [Python 3.13 is just around the corner](https://www.python.org/downloads/release/python-3130rc2/#:~:text=This%20release%2C%203.13.0rc2%2C,%2C%202024%2D10%2D01.), should DRF wait for it to release 3.16? 

I think support could be added in 3.16.1 (assuming no major changes), as Django sometimes add support to earlier versions, when the effort is acceptable.

EDIT: addressed in https://github.com/encode/django-rest-framework/pull/9527

**Comment by browniebroke** [comment]
What is going to be the branching strategy for 3.15.3? I see 2 options:

1. Branch off 3.15.2 as e.g. `stable/3.15.x` (same as Django) to cherry pick the changes from `master` 
2. Release it directly from `master`

There are a few changes in `master` that may be better of saving for 3.16, and reinstating older Python/Django versions (IMO) might be easier off `master`: no need to remove them again. Thoughts? A third option I missed?

**Reply by lovelydinosaur** [reply]
To date we've just released everything from `master`.
Right now I'd suggest we don't make any changes there.

(Also. `master` Ugh. Let's switch up to `main` sometime...)
